Dependencies/xr/Source/OpenXR/SceneUnderstanding.cpp

```diff
@@ -0,0 +1,497 @@
+#define NOMINMAX // avoid including macro max definitions in favor of limit max definitions
+
+#include "SceneUnderstanding.h"
+#include "XrRegistry.h"
+
+using namespace xr;
+
+constexpr SceneUnderstanding::DetectionBoundaryType DEFAULT_BOUNDARY_TYPE = SceneUnderstanding::DetectionBoundaryType::Sphere;
+constexpr float DEFAULT_SPHERE_RADIUS = 5.f;
+constexpr double DEFAULT_UPDATE_INTERVAL_IN_SECONDS = 2;
+constexpr double NANOSECONDS_IN_SECOND = 1000000000;
+
+inline xr::Pose XrPoseToBabylonPose(XrPosef pose)
+{
+    // Right handed to left handed conversions are handled in the Babylon.js layer
+    return xr::Pose{
+        {
+            pose.position.x,
+            pose.position.y,
+            pose.position.z,
+        },
+        {
+            pose.orientation.x,
+            pose.orientation.y,
+            pose.orientation.z,
+            pose.orientation.w
+        }};
+}
+
+SceneUnderstanding::SceneUnderstanding::InitOptions::InitOptions(

```

Comment:
Should this class be called `XrSceneUnderstanding` for consistency?

Reply:
I'm not sure if long term that will generate more confusion/lead folks to believe that this type is an OpenXR type. But, I also don't mind making this change/will go a head and update things.

Reply:
Actually, I think we should keep this as SceneUnderstanding for now. We don't want to rename the header to XrSceneUnderstanding, because thats already used by the OpenXR-MixedReality dependency.